### PR TITLE
Fix broken link for doc_source_url

### PR DESCRIPTION
**Pull request by jeongseok-meta**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@jeongseok-meta thanks for the PR but we really needed to get #121 in first. We can add the URL update as soon as the bot re-issues #132.

**Comment by jeongseok-meta** [comment]
@ocefpaf Sounds good. I've rebased and rerendered this PR, but feel free to discard it in favor of reissue of #133.

**Comment by h-vetinari** [comment]
No reason for a bot rerun if we have a valid PR already (IMO, though not a maintainer here), however, we do need to fix some fallout from #121 (can be separately or in this PR directly from my POV). Please see the post-merge review there, it's an easy fix.

**Comment by jeongseok-meta** [comment]
Updated to keep windows compatible with ICU following the first comment

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ cxx_compiler:      # [osx]
 - clang_bootstrap  # [osx]
 
 with_icu:
-- yes     # [not win]
```
Comment (ocefpaf) [inline recipe/conda_build_config.yaml:7]: @isuruf can you comment on this here? I recall that you added this for a reason, right?

Comment (h-vetinari) [reply recipe/conda_build_config.yaml:7]: It's fine to keep that part as-is, but then we need to not apply the `icu <0.0` run-constraint on windows. Here's a concrete [example](https://github.com/conda-forge/libpdal-feedstock/pull/17) that fails due the constraint.

**Comment by jakirkham** [comment]
Recently the bot opened PR: https://github.com/conda-forge/libxml2-feedstock/pull/135

Looks like one of the maintainers saw that notification, but missed this one. So had merged that

As this recreated the Windows issue alluded to above, a separate fix was made right after in PR: https://github.com/conda-forge/libxml2-feedstock/pull/136

Think that subsumed a lot of the changes here

<hr>

Sorry this PR was missed 😞  Do appreciate you putting in time to update this Jeongseok

It does look like there are still some changes here that would be nice to include (for example fixing the `doc_source_url`, which is currently broken). Would you be willing to merge or rebase this PR? Then we can focus on getting those remaining fixes in

**Comment by jeongseok-meta** [comment]
Not a problem. It's great to see the active updates from the maintainers! After rebasing, I found that it was a very small fix. Thank you for providing the context, @jakirkham!

**Comment by jakirkham** [comment]
Thanks Jeongseok! 🙏

Appreciate your understanding and willingness to work with us 🙂

**Comment by ocefpaf** [comment]
@jeongseok-meta thanks for hanging in there. This feedstock was kind of behind and we all jumped to update it at the same time.

**Pull request merged**